### PR TITLE
SI-7376 Scaladoc warns when discarding local doc comments with API tags

### PR DESCRIPTION
**Pull request by som-snytt**
Double-star doc comments in non-dockable positions at the end of a block
will emit a warning only if API tags like @author are present, or under
-Xlint.

A real comment parser is applied early to probe for tags, to minimize
ad hoc testing or duplication, but warnings are suppressed. Residual
ad hockiness lies in precisely which tags to warn on.  Ad hoc or ad doc.

This fix is a stop gap; a richer solution would also report about other
doc locations that won't be processed.

Review by @paulp, of whose idea this is a refinement.  Of whom?

I'll also take advice from Mr Thor @author.  Not sure whom best to impose upon.  Upon whom?

### CODE REVIEW

**Comment by som-snytt** [comment]
I did run test.scaladoc first this time.

**Comment by som-snytt** [comment]
I did run docs.comp first this time.  Actually, I was already running it, but I had an itchy push finger.

File: test/scaladoc/run/t5527.scala [hunk 1]
```diff
@@ -49,6 +49,47 @@ object Test extends DirectTest {
...
           false
       }
+
+      def test5 {
+        /** @martin is this right? It shouldn't flag me as scaladoc. */
+        if (true) ???
+      }
+
+      def test6 = {
+        /** Document this crucial constant for posterity.
```
Comment (paulp) [inline test/scaladoc/run/t5527.scala:59]: u=4. That's sufficient explanation. Rederive it, people.

**Comment by paulp** [comment]
LGTM but I admit my interest in the subject matter is not setting any records.

**Comment by som-snytt** [comment]
Kind of intentional to be uninteresting.  But thanks for your time.

I'm trying to start useful review; it seems that uninteresting should mean community review. If there were a way to flag that, maybe it would help off-load the work, especially for what is less interesting.

Kind of sorry Mr A. U. Thor didn't step up to the plate.

**Pull request merged**